rta/src/main/java/com/gluonhq/richtextarea/viewmodel/RichTextAreaViewModel.java

```diff
@@ -735,10 +740,12 @@ private void updateProperties() {
         redoStackSizeProperty.set(commandManager.getRedoStackSize());
     }
 
-    private Document getCurrentDocument() {
+    private Document getCurrentDocument(Selection selection) {
         // text and indices should be based on the exportable text
         int caret = getTextBuffer().getText(0, getCaretPosition()).length();
-        return new Document(getTextBuffer().getText(), getTextBuffer().getDecorationModelList(), caret);
+        int start = selection.isDefined() ? selection.getStart() : 0;
+        int end = selection.isDefined() ? selection.getEnd() : getTextLength();
+        return new Document(getTextBuffer().getText(start, end), getTextBuffer().getDecorationModelList(start, end), caret);
```

Comment:
There are 3 calls to `getTextBuffer()` in this method. Wondering if we should instead create a variable.

Reply:
Not needed. In this class `getTextBuffer()` it is used 30+ times, without extracting variables. After all, it is just a getter from a property, with a null check.